File: service/app/src/main/java/io/miragon/blueprint/adapter/inbound/operaton/SendContractDelegate.java
package io.miragon.blueprint.adapter.inbound.operaton;

import io.miragon.blueprint.application.port.inbound.SendContractUseCase;
import io.miragon.blueprint.domain.leasing.ApplicationId;
import org.operaton.bpm.engine.delegate.DelegateExecution;
import org.springframework.stereotype.Component;

@Component
public class SendContractDelegate extends BaseDelegate {

    private final SendContractUseCase useCase;

    public SendContractDelegate(SendContractUseCase useCase) {
        this.useCase = useCase;
    }

    @Override
    protected void executeTask(DelegateExecution execution) {
        useCase.sendContract(ApplicationId.of(execution.getProcessBusinessKey()));
    }
}

File: service/app/src/main/kotlin/io/miragon/blueprint/adapter/inbound/operaton/SendContractDelegate.kt
package io.miragon.blueprint.adapter.inbound.operaton

import io.miragon.blueprint.application.port.inbound.SendContractUseCase
import io.miragon.blueprint.domain.leasing.ApplicationId
import org.operaton.bpm.engine.delegate.DelegateExecution
import org.springframework.stereotype.Component

@Component
class SendContractDelegate(
    private val useCase: SendContractUseCase,
) : BaseDelegate() {

    override fun executeTask(execution: DelegateExecution) {
        useCase.sendContract(ApplicationId.of(execution.processBusinessKey))
    }
}
